clients/drcachesim/reader/reader.h

```diff
@@ -171,9 +196,14 @@ class reader_t : public std::iterator<std::input_iterator_tag, memref_t>,
     const char *output_prefix_ = "[reader]";
     uint64_t cur_ref_count_ = 0;
     uint64_t cur_instr_count_ = 0;
-    uint64_t chunk_instr_count_ = 0; // Unchanging once set to non-zero.
     uint64_t last_timestamp_instr_count_ = 0;
     trace_entry_t *input_entry_ = nullptr;
+    // Remember top-level headers for the memtrace_stream_t interface.
+    uint64_t version_ = 0;
+    uint64_t filetype_ = 0;
+    uint64_t cache_line_size_ = 0;
+    uint64_t chunk_instr_count_ = 0; // Unchanging once set to non-zero.
```

Comment:
The "Unchanging..." comment is true for other fields too right? It's confusing that it's mentioned only for one.